lib/ts/querier.ts

```diff
@@ -49,11 +55,12 @@ export default class Querier {
             postAPIHook
         );
 
+        const fetchResponse = result.clone();
```

Comment:
Instead of cloning here in all the APIs, consider cloning inside `getResponseJsonOrThrowGeneralError`:
```ts
getResponseJsonOrThrowGeneralError = async (response: Response): Promise<any> => {
        let json = await response.clone().json();

        if (json.status === "GENERAL_ERROR") {
            let message = json.message === undefined ? "No Error Message Provided" : json.message;
            throw new Error(message);
        }

        return json;
    };
```

Reply:
Done